modules/@apostrophecms/doc-type/index.js

```diff
@@ -502,6 +503,23 @@ module.exports = {
         }
         return self.apos.migration.addSortify(self.__meta.name, { type: self.name }, field);
       },
+      // Add the "cacheInvalidatedAt" field to the documents that do not have it yet,
+      // and set it to equal doc.updatedAt.
+      addCacheFieldMigration() {
+        self.apos.migration.add(`add-cache-invalidated-at-field-for-${self.__meta.name}`, () => {
+          self.apos.migration.eachDoc({ type: self.__meta.name }, 5, async doc => {
```

Comment:
This is the right place to do `cacheInvalidatedAt: { $exists: 0 }`. Doing that in `updateOne` means we pay for an update call for every doc, whether it already has this property or not.